### ISSUE TITLE
Highlight on keys not working in Rider

### ISSUE THREAD

**Issue by piskov** [joshestein/ideavim-quickscope#15]
I’ve just recently came across this plugin — love the idea, but it seems not working in Rider.

Here is the top part of my config:
```
source ~/.vimrc

" plugins
set argtextobj
set commentary
set highlightedyank
set matchit
set multiple-cursors
set NERDTree
set textobj-indent
set vim-paragraph-motion

" ideavim settings
set ideajoin
set ideastatusicon=gray
set peekaboo
set visualbell

" quickscope
set quickscope
let g:qs_highlight_on_keys = ['f', 'F', 't', 'T']
```

And top part of .vimrc:
```
" plugins
set easymotion
set ReplaceWithRegister
set surround

" line numbers
set number
set relativenumber

" sensible search
set hlsearch
set ignorecase
set incsearch
set smartcase
```

Despite having `qs_highlight_on_keys` set, it seems to have a broken logic

1. Here we start in normal mode and highlight is always shown:

![image](https://github.com/user-attachments/assets/61c2943d-3ad7-4cae-bdb3-80907b819301)

2. Now I press `i` to go to insert mode, but highlight is still on:

![image](https://github.com/user-attachments/assets/228bddfd-2fb6-47b6-882f-421fb9663fcf)

4. It only hides if I press one more key after I’m already in insert mode:

![image](https://github.com/user-attachments/assets/e91772d5-a161-4cfb-9fd1-a51abfad60ee)

5. If I then press Esc and go to normal mode, there is no hightlight (good):

![image](https://github.com/user-attachments/assets/f15c54b8-854c-402a-9537-aa050b8e2aa9)

6. But the moment I press another key in normal mode, hightlight is back (bad):

![image](https://github.com/user-attachments/assets/357e32f1-7938-4469-b09c-8367f9eef04f)

—

tldr;
1. At least in Rider `qs_highlight_on_keys ` is not working (hightlight is always on in normal mode).
2. Highlight is still upon entering Insert mode unless another key is pressed.

The versions I have: 2024.3 and 1.0.12

**Comment by piskov** [comment]
On a whim tried to swapping the lines (let before set) and restart — now it works:

```
let g:qs_highlight_on_keys = ['f', 'F', 't', 'T']
set quickscope
```
